confgenerator/testdata/goldens/metrics-receiver_prometheus/golden/linux-gpu/otel.yaml

```diff
@@ -525,7 +525,7 @@ receivers:
             __meta_gce_instance_name: test-instance-name
             __meta_gce_interface_ipv4_nictest_interface: test-interface-ipv4
             __meta_gce_machine_type: test-machine-type
-            __meta_gce_metadata_test_escape: $${foo:bar}
+            __meta_gce_metadata_test_escape: _{foo:bar}
```

Comment:
Do we need to add another test case for this newly added sanitizing logic? So that we test both `$ -> $$` and `${ -> _{`:
```
__meta_gce_metadata_test_escape: $$foo  <- when there is $foo in the metadata
__meta_gce_metadata_test_sanitize: _{foo:bar}
```

Reply:
Added another metadata field that uses `$foo` to confirm it generates `$$foo`. Done